internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -95,7 +95,7 @@ class JavaCompilerSpec extends UnitSpec {
         val (result, _) = compile(local, Seq(input), Seq("-d", out.getAbsolutePath))
         result shouldBe true
         val clazzz = new URLClassLoader(Array(out.toURI.toURL)).loadClass("hasstaticfinal")
-        ClassToAPI(Seq(clazzz))
+        ClassToAPI(Seq(clazzz)).sortBy(_.definitionType.hashCode)
```

Comment:
> Removing laziness from the fields of ClassLike has the side effect to randomize the order of the result from compileWithPrimitive() method (which returns a Seq containing a class and an object).
  
Can you diagnose why? This is strange.
  
> Sorting the result ensure we have an order consistent with the expected result of the test.
  
In my opinion, we should not sort here just for testing purposes. Our core logic does not require it, it has no effect on Zinc changes analysis and it's not a contract of our API. Can we solve it in the tests side?